system.go

```diff
@@ -30,6 +30,16 @@ import (
 	_ "github.com/elastic/go-sysinfo/providers/windows"
 )
 
+type ProviderOption func(*registry.ProviderOptions)
+
+// WithHostFS returns a provider with a custom HostFS root path,
+// enabling use of the library from within a container, or an alternate root path.
```

Comment:
I think this should go into more detail to help ensure it is used correctly. I think it should give an example of what mounts are needed.

```
// For example, WithHostFS("/hostfs") could be used when these mounts are present
// in the container
//
//   - /proc:/hostfs/proc
//   - /sys/fs/cgroup:/hostfs/sys/fs/cgroup
//   - ...
```